pkg/reconciler/contour/config/contour.go

```diff
@@ -54,9 +58,13 @@ type visibilityValue struct {
 // NewContourFromConfigMap creates an Contour config from the supplied ConfigMap
 func NewContourFromConfigMap(configMap *corev1.ConfigMap) (*Contour, error) {
 	var tlsSecret *types.NamespacedName
+	var timeoutPolicyResponse = "infinity"
+	var timeoutPolicyIdle = "infinity"
 
 	if err := configmap.Parse(configMap.Data,
 		configmap.AsOptionalNamespacedName(defaultTLSSecretConfigKey, &tlsSecret),
+		configmap.AsString(timeoutPolicyResponseKey, &timeoutPolicyResponse),
+		configmap.AsString(timeoutPolicyIdleKey, &timeoutPolicyIdle),
```

Comment:
I believe these can either be `"infinite"` or a `time.Duration`.  We should probably validate this lest we turn all of the configmaps in the cluster invalid 😬

Reply:
Missed this

Reply:
Contour spec accepts string only. Is there any similar example validation I can look at?

Reply:
Right, but I think other things will fail.

You might at `asContourDuration` in the style of this, which validates that the value is "infinity" or `ParseDuration` doesn't return an error: https://github.com/knative/pkg/blob/2465d13e724265add1a6bce0a8f71564e6c1d1cf/configmap/parse.go#L112-L124

Reply:
So I can add a function called `asContourDuration` in `knative/pkg/configmap/parse.go` or a function somewhere in net-contour should do this validation?

Reply:
I'd just add a private function here that does it.

Reply:
Okay..Done some changes.PTAL